be/src/pipeline/exec/olap_scan_operator.cpp

```diff
@@ -601,9 +601,17 @@ Status OlapScanLocalState::prepare(RuntimeState* state) {
     }
 
     for (size_t i = 0; i < _scan_ranges.size(); i++) {
-        RETURN_IF_ERROR(_tablets[i].tablet->capture_rs_readers({0, _tablets[i].version},
-                                                               &_read_sources[i].rs_splits,
-                                                               _state->skip_missing_version()));
+        if (config::is_cloud_mode() && _state->enable_query_freshness_tolerance()) {
+            RETURN_IF_ERROR(std::static_pointer_cast<CloudTablet>(_tablets[i].tablet)
+                                    ->capture_rs_readers_with_freshness_tolerance(
+                                            {0, _tablets[i].version}, &_read_sources[i].rs_splits,
+                                            _state->skip_missing_version(),
+                                            _state->query_freshness_tolerance_ms()));
+        } else {
+            RETURN_IF_ERROR(_tablets[i].tablet->capture_rs_readers({0, _tablets[i].version},
+                                                                   &_read_sources[i].rs_splits,
+                                                                   _state->skip_missing_version()));
+        }
```

Comment:
Maybe we can add a function capute_rs_readers one of whose arguments is state and then move if here into mew capute_rs_readers.